vowpalwabbit/core/include/vw/core/global_data.h

```diff
@@ -145,38 +217,23 @@ class workspace
   // May be nullptr, so you must check before calling it
   std::function<void(float)> set_minmax;
 
-  uint64_t current_pass;
+  details::passes_config pc;

```

Comment:
Since we dont want these to change - would could only expose these to the overall library via const ref and require using a getter function

Reply:
change is interesting, passes for example keeps state of current pass. in this refactor I haven't put much thought on next steps, for now it's about classification and seeing where these sub structs get used with help of grep and the short name